### DISCUSSION TITLE
extending lsp configuration for a given language (groovy)

### DISCUSSION THREAD

**Discussion by gtnbssn** [LazyVim/LazyVim#4275] [Q&A] [answered]
I am trying to setup the groovy lsp properly, and it seems this requires explicitly pointing the server to the location of the dependencies:
https://www.reddit.com/r/neovim/comments/138m3l0/help_setting_up_groovy_lsp/

(It seems weird to me but I am a complete beginner with java and groovy sooo...)

How would I go about doing this with lazyvim?

**Comment by dpetka2001** [comment, marked as answer]
```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      groovyls = {
        settings = {
          groovy = {
            classpath = {
              "/path/where/the/files/are/located/lib",
              "/second/path/where/the/files/are/located/lib",
            },
          },
        },
      },
    },
  },
},
```
Is how you would configure a LSP server on LazyVim, but I don't do any Java/Groovy so not entirely sure.

**Reply by gtnbssn** [reply]
Thanks! I will try to figure out the java/groovy part now.

**Reply by gtnbssn** [reply]
This seems to mean I need to setup the path for each and every project in my editor's setting though. Isn't that weird?

**Reply by dpetka2001** [reply]
No idea. Like i said i've no experience with Java/Groovy projects.

**Reply by rogeriofrsouza** [reply]
Hum, I don't know much about Groovy but according to [Groovy LSP repo](https://github.com/GroovyLanguageServer/groovy-language-server)

> groovy.classpath (string[] - sets a custom classpath to include .jar files)

Maybe there is a default classpath, and you only set a custom one depending on your needs 🤔
